src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -3347,22 +3336,13 @@ private void login(String primary, String primaryInstanceName, int primaryPortNu
                 }
 
                 int errorCode = e.getErrorCode();
-                int driverErrorCode = e.getDriverErrorCode();
+                driverErrorCode = e.getDriverErrorCode();
                 sqlServerError = e.getSQLServerError();
 
-                if (SQLServerException.LOGON_FAILED == errorCode // logon failed, ie bad password
-                        || SQLServerException.PASSWORD_EXPIRED == errorCode // password expired
-                        || SQLServerException.USER_ACCOUNT_LOCKED == errorCode // user account locked
-                        || SQLServerException.DRIVER_ERROR_INVALID_TDS == driverErrorCode // invalid TDS
-                        || SQLServerException.DRIVER_ERROR_SSL_FAILED == driverErrorCode // SSL failure
-                        || SQLServerException.DRIVER_ERROR_INTERMITTENT_TLS_FAILED == driverErrorCode // TLS1.2 failure
-                        || SQLServerException.DRIVER_ERROR_UNSUPPORTED_CONFIG == driverErrorCode // unsupported config
-                                                                                                 // (eg Sphinx, invalid
-                                                                                                 // packetsize, etc)
-                        || (SQLServerException.ERROR_SOCKET_TIMEOUT == driverErrorCode // socket timeout
-                                && (!isDBMirroring || attemptNumber > 0)) // If mirroring, only close after failover has been tried (attempt >= 1)
-                        || timerHasExpired(timerExpire)
-                // for non-dbmirroring cases, do not retry after tcp socket connection succeeds
+                if (shouldFailConnection(errorCode, driverErrorCode, isDBMirroring, useTnir, attemptNumber,
```

Comment:
If a transient error were to happen here, it looks like it would immediately terminate the connection. Prior, the logic was that if a transient error occurred, the driver would attempt another connection.

If a redirection failure were to happen, and a transient error was thrown, the driver wouldn't go back to the GW to attempt another connection.